### ISSUE TITLE
Documentation is out of sync with the latest release

### ISSUE THREAD

**Issue by ccapndave** [darklang/tablecloth#171]
The documentation shows functions such as `Result.ok`, `Result.error` and `Option.both` which don't actually exist in `0.0.7`.  It would be nice if the documentation had a version selector (maybe similar to Elixir's docs - https://hexdocs.pm/phoenix/Phoenix.html) so that you can choose `master` or `0.0.7`.

On a side note I would like to use the functions in `master`, but I can't figure out how to get them working.  Putting `"tablecloth-bucklescript": "darklang/tablecloth#master"` into `package.json` doesn't seem to work.  Any advice would be appreciated!

**Comment by Dean177** [comment]
The next release will happen really soon, I just need to get around to writing the changelog. 

I version selector is a good idea, if we couple adding a new functions with a `since x.y.z` annotation in the docs it shouldn't be too much work to implement either, can you create a separate issue for the feature request?

Are you getting any specific errors when trying to use the latest master?

The  `integration-test` folder always gets built and run against master, maybe that would be useful to check your configuration against?

**Comment by ccapndave** [comment]
Sure!  https://github.com/darklang/tablecloth/issues/172

The whole thing just failed to build with a lot of strange errors (`Cannot find module Tablecloth`, `Cannot find module React`).  Do you think it should work if I add `"tablecloth-bucklescript": "darklang/tablecloth#master"` to the dependencies?

**Comment by Dean177** [comment]
The format looks correct to me.

Perhaps try `bsb -clean-world` followed by `bob -make-world`

**Comment by ccapndave** [comment]
I had another go, and I think the problem is that there is a compile error that is preventing Tablecloth from compiling, which then causes subsequent errors:

```
  Warning number 3 (configured as error)
  /Users/dave/Projects/Couloir/ctp/title-cp/node_modules/tablecloth-bucklescript/bucklescript/src/TableclothInt.ml 45:29-43

  43 │ let ( /. ) n by = Js.Int.toFloat n /. Js.Int.toFloat by
  44 │
  45 │ let power ~base ~exponent = Js.Math.pow_int ~base ~exp:exponent
  46 │
  47 │ let ( ** ) base exponent = Js.Math.pow_int ~base ~exp:exponent

  deprecated: Js.Math.pow_int
use `power_float` instead, the return type may be not int

  Warning number 3 (configured as error)
  /Users/dave/Projects/Couloir/ctp/title-cp/node_modules/tablecloth-bucklescript/bucklescript/src/TableclothInt.ml 47:28-42

  45 │ let power ~base ~exponent = Js.Math.pow_int ~base ~exp:exponent
  46 │
  47 │ let ( ** ) base exponent = Js.Math.pow_int ~base ~exp:exponent
  48 │
  49 │ let negate = ( ~- )

  deprecated: Js.Math.pow_int
use `power_float` instead, the return type may be not int
```

**Comment by Dean177** [comment]
It looks like you might be using a globally installed version of bsb? 

We don't support anything more than `7.2.2` at the moment: https://github.com/darklang/tablecloth/issues/141 

Try using the locally installed version using the scripts in `package.json`

```
npm run clean
npm run build
```

**Comment by ccapndave** [comment]
Does that mean that my application has to use 7.2.2 too, or can Tablecloth be compiled with 7.2.2 but I can still use it in an 8.1.1 app?

**Comment by Dean177** [comment]
I means that your app has to use 7.2.2 too.

#141 is the only issue stopping us from supporting a newer version of Bucklescript I think.

**Comment by ccapndave** [comment]
Ah ok - I'm keen to change over to the new Bucklescript syntax so I'll stay on 8.2.2 and wait for a Tablecloth update in that case.  Cheers!

**Comment by danielo515** [comment]
So all the documentation that I have been reading for more than one hour is basically useless?  I want to use the Map.String module and I'm getting an error telling me that such module does not exist, then I checked the mli files and indeed there is no definition for that.

**Comment by pbiggar** [comment]
Apologies @danielo515, we need to get a release out. Can you use git master directly in your app?

**Comment by danielo515** [comment]
Don't worry @pbiggar, I'm using reason standard for now. Since most of the docs seems to come from there it is probably a good substitute for many. I'll be looking for the release (with standard merged in)

**Comment by pbiggar** [comment]
Actually released 0.0.8 and the website matches.

**Comment by ccapndave** [comment]
This is so weird - I was working on this project today for the first time in 2 years!
